Assets/Cannon.cs

```diff
@@ -0,0 +1,51 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Cannon : MonoBehaviour
+{
+    public List<GameObject> Targets;
+    public GameObject ProjectilePrefab;
+    public float ProjectileVelocity;
+    public float ReloadTime;
+    public Vector3 ShootDirection;
+    public float ShootAngle;
+    private float LastShootTime = 0.0f;
+    
+    void Start()
+    {
+        LastShootTime = (float)Time.timeAsDouble;
+    }
+
+    void Update()
+    {
+        if ((float)Time.timeAsDouble >= LastShootTime + ReloadTime) {
+            LastShootTime = (float)Time.timeAsDouble;
+            GameObject CurrentTarget = null;
+            foreach (GameObject GO in Targets) {
+                if (GO && 
+                    Vector3.Dot(ShootDirection, 
+                                Vector3.Normalize(GO.transform.position - transform.position)
+                    ) > Mathf.Cos(Mathf.Deg2Rad * ShootAngle)) {
+                    if (GO.GetComponent<RagDoll>() && 
```

Comment:
Repeating component access is inefficient, you should save the result into a local variable and then work with it